Source/CombatExtended/CombatExtended/Projectiles/ProjectileCE_Bursting.cs

```diff
@@ -0,0 +1,72 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using RimWorld;
+using UnityEngine;
+using Verse;
+using Verse.Sound;
+using CombatExtended.Compatibility;
+using CombatExtended.Lasers;
+using ProjectileImpactFX;
+using CombatExtended.Utilities;
+
+namespace CombatExtended
+{
+    class ProjectileCE_Bursting : ProjectileCE
+    {
+        private int ticksToBurst;
+
+        public override void ExposeData()
+        {
+            base.ExposeData();
+            Scribe_Values.Look(ref this.ticksToBurst, "ticksToBurst", -1, false);
+        }
+
+        public override void Launch(Thing launcher, Vector2 origin, float shotAngle, float shotRotation, float shotHeight = 0f, float shotSpeed = -1f, Thing equipment = null, float distance = -1)
+        {
+            int armingDelay = 0;
+            if (def.projectile is ProjectilePropertiesCE props) armingDelay = props.armingDelay;
+            if (distance > 0)
+            {
+                float cosine = (float)Math.Cos((double)shotAngle);
+
+                float fuzeTiming = distance / ((shotSpeed / 60) * cosine);
+                //Log.Message("Distance    = " + distance);
```

Comment:
If you want to leave these in, put them *functional* inside an `#if DEBUG` block.  Otherwise, let's not litter comments in production code.